tests/unit/controllers/api/v1/test_kvps_rbac.py

```diff
@@ -41,360 +47,605 @@ def setUp(self):
+        permission_grants = [str(grant_5_db.id), str(grant_6_db.id)]
+        role_2_db = RoleDB(name="user5", permission_grants=permission_grants)
+        role_2_db = Role.add_or_update(role_2_db)
+        self.roles["user_5"] = role_2_db
+
+        grant_7_db = PermissionGrantDB(
+            resource_uid="key_value_pair:st2kv.user:user6:test_new_key_3",
+            resource_type=ResourceType.KEY_VALUE_PAIR,
+            permission_types=[PermissionType.KEY_VALUE_SET],
+        )
+        grant_7_db = PermissionGrant.add_or_update(grant_7_db)
+        grant_8_db = PermissionGrantDB(
+            resource_uid="key_value_pair:st2kv.user:user6:test_new_key_3",
+            resource_type=ResourceType.KEY_VALUE_PAIR,
+            permission_types=[PermissionType.KEY_VALUE_VIEW],
+        )
+        grant_8_db = PermissionGrant.add_or_update(grant_8_db)
+        permission_grants = [str(grant_7_db.id), str(grant_8_db.id)]
+        role_1_db = RoleDB(name="user6", permission_grants=permission_grants)
+        role_1_db = Role.add_or_update(role_1_db)
+        self.roles["user_6"] = role_1_db
+
+        # Role assignments
+        user_db = self.users["user_2"]
+        role_assignment_db = UserRoleAssignmentDB(
+            user=user_db.name,
+            role=self.roles["user_2"].name,
+            source="assignments/%s.yaml" % user_db.name,
+        )
+        UserRoleAssignment.add_or_update(role_assignment_db)
+
+        user_db = self.users["user_1"]
+        role_assignment_db = UserRoleAssignmentDB(
+            user=user_db.name,
+            role=self.roles["user_1"].name,
+            source="assignments/%s.yaml" % user_db.name,
+        )
+        UserRoleAssignment.add_or_update(role_assignment_db)
+
+        user_db = self.users["user_3"]
+        role_assignment_db = UserRoleAssignmentDB(
+            user=user_db.name,
+            role=self.roles["user_3"].name,
+            source="assignments/%s.yaml" % user_db.name,
+        )
+        UserRoleAssignment.add_or_update(role_assignment_db)
+
+        user_db = self.users["user_4"]
+        role_assignment_db = UserRoleAssignmentDB(
+            user=user_db.name,
+            role=self.roles["user_4"].name,
+            source="assignments/%s.yaml" % user_db.name,
+        )
+        UserRoleAssignment.add_or_update(role_assignment_db)
+
+        user_db = self.users["user_5"]
+        role_assignment_db = UserRoleAssignmentDB(
+            user=user_db.name,
+            role=self.roles["user_5"].name,
+            source="assignments/%s.yaml" % user_db.name,
+        )
+        UserRoleAssignment.add_or_update(role_assignment_db)
+
+        user_db = self.users["user_6"]
+        role_assignment_db = UserRoleAssignmentDB(
+            user=user_db.name,
+            role=self.roles["user_6"].name,
+            source="assignments/%s.yaml" % user_db.name,
+        )
+        UserRoleAssignment.add_or_update(role_assignment_db)
```

Comment:
There are a lot of users, roles, and grants added here. Are these test cases all covered in the unit tests below?  It's hard to follow. Can you add comments what these users/roles/grants are used in which tests?